docs/advanced/tls_hardware_acceleration.md

```diff
@@ -127,8 +129,24 @@ To replace a module you have to:
 
 - Provide a header file for your implementation. The file name must be `<Module Name>_alt.h`.
 
-- Set the macro `MBEDTLS_<Module Name>_ALT` to notify mbed TLS and the compiler or linker about the replacement. You can read more on this in the [subsection about setting macros](#How-to-set-the-macros).
+- Set the macro `MBEDTLS_<Module Name>_ALT` to notify mbed TLS and the compiler or linker about the replacement. You can read more on this in the [subsection about setting macros](#how-to-set-the-macros).
 
 ### Where to find the default implementations
 
 The default implementation of the modules are usually in the file `feature/mbedtls/src/<Module Name>.c`. The ECP module is split to two files: `ecp.c` and `ecp_curves.c`.
+
+## Considerations on alternative implementations
+
+### Concurrency
+
+Note that functions in mbed TLS can be called from multiple threads and from multiple processes at the same time. As hardware accelerators are usually a unique resource, it is important to protect all functions against concurrent access.
+
+For short actions, disabling interrupts for the duration of the operation may be enough. When it is not desirable to prevent context switches during the execution of the operation, take care to protect the operation with mutual exclusion primitive such as a [mutex](https://docs.mbed.com/docs/mbed-os-api-reference/en/latest/APIs/tasks/rtos/#mutex). Make sure to unlock the mutex or restore the interrupt status when returning from the function even if an error occurs.
+
+### Power management
+
+The current framework does not provide an interface to initialize and shut down accelerator hardware. A simple approach is to perform any necessary hardware initialization during system startup (outside of mbed TLS), however this may not be desirable for power consumption reasons. At the other end of the spectrum, it is possible to initialize the hardware at the beginning of each function, and shut it down as soon as the results have been read. This is a viable strategy if initialization is cheap enough.
```

Comment:
- Please replace "A simple" with "One" in the first sentence.
- Please replace "TLS), however this" with "TLS); however, this" for proper semicolon use.
- Please delete the comma after "function" in the last sentence to avoid the comma splice.
- Query: Who or what is reading the results? Please clarify for active voice.